### PR TITLE
Clear the build directory before building

### LINKED ISSUE #193: Add clean task before build
Right now, running `polymer build` does not seem to clean out the build directory before building. 

I started with the app-drawer-template example and after replacing the my-view* files with my own files, I could see that all the old files are still present in the build directory.

### PR DESCRIPTION
**Pull request by FredKSchott**
Resolves #193
Inspired by @chuckh's work on https://github.com/PolymerElements/polymer-starter-kit/pull/886

/cc @justinfagnani

### CODE REVIEW

File: src/build/build.ts [hunk 1]
```diff
@@ -20,12 +21,13 @@ import * as logging from 'plylog';
 import {PolymerProject, addServiceWorker, forkStream, DepsIndex, SWConfig} from 'polymer-build';
 
 import {JSOptimizeStream, CSSOptimizeStream, HTMLOptimizeStream} from './optimize-streams';
-
 import {ProjectConfig} from '../project-config';
 import {PrefetchTransform} from './prefetch';
 import {waitFor} from './streams';
 import {parsePreCacheConfig} from './sw-precache';
 
+const BUNDLED_BUILD_DIR = path.join('build', 'bundled');
```
Comment (FredKSchott) [inline src/build/build.ts]: @justinfagnani just for you 😜

Comment (justinfagnani) [inline src/build/build.ts]:
```javascript
const STOP_YELLING_AT_ME = true;
```

Comment (justinfagnani) [inline src/build/build.ts]: seriously though, I just don't think the caps are useful in any way.

File: src/build/build.ts [hunk 2]
```diff
@@ -48,28 +50,31 @@ export interface BuildOptions {
...
-    // mix in optimization options from build command
-    // TODO: let this be set by the user
-    let optimizeOptions = {
-      html: Object.assign({removeComments: true}, options.html),
-      css: Object.assign({stripWhitespace: true}, options.css),
-      js: Object.assign({minify: true}, options.js),
-    };
+  logger.info(`Cleaning build/ directory...`);
 
+  return <Promise<void>>del([UNBUNDLED_BUILD_DIR, BUNDLED_BUILD_DIR]).then(() => {
```
Comment (justinfagnani) [inline src/build/build.ts]: why do you need the cast?

File: src/build/build.ts [hunk 3]
```diff
@@ -48,28 +50,31 @@ export interface BuildOptions {
...
+    logger.debug(`build/ directory cleaned.`);
```
Comment (justinfagnani) [inline src/build/build.ts]: maybe this should be an info?

File: src/build/build.ts [hunk 4]
```diff
@@ -134,17 +139,13 @@ export function build(options: BuildOptions, config: ProjectConfig): Promise<any
...
       });
     });
 
-    return Promise.all([unbundledPostProcessing, bundledPostProcessing]).then(() => {
-      logger.info('Build complete!');
-      buildResolve();
-    });
-
-  });
+    return Promise.all([unbundledPostProcessing, bundledPostProcessing]);
```
Comment (justinfagnani) [inline src/build/build.ts]: convert this to an `async` function, and you will be happier :)

**Comment by justinfagnani** [comment]
@FredKSchott what's up w/ this guy?

**Comment by FredKSchott** [comment]
very unloved, I have a merge-conflict-fix branch floating around somewhere on my machine...

**Comment by justinfagnani** [comment]
pingy ping

**Comment by FredKSchott** [comment]
@justinfagnani rebased and fixed up, PTAL

**Pull request merged**